### PR TITLE
Update cmake for optional non-local install

### PR DESCRIPTION
**Pull request by xylar**
This is needed to support conda-forge

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,17 +1,37 @@
...
+endif (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+
 add_executable (jigsaw-cmd jigsaw.cpp)
 target_compile_definitions (jigsaw-cmd PRIVATE __cmd_jigsaw)
 set_target_properties (jigsaw-cmd PROPERTIES OUTPUT_NAME jigsaw)
-install (TARGETS jigsaw-cmd DESTINATION "${PROJECT_SOURCE_DIR}/bin")
+if (DEFINED INSTALL_LOCAL)
+  install (TARGETS jigsaw-cmd DESTINATION "${PROJECT_SOURCE_DIR}/bin")
+else ()
+  install (TARGETS jigsaw-cmd DESTINATION bin)
```
Comment (xylar) [inline src/CMakeLists.txt]: @dengwirda, on conda-forge it doesn't work to install locally to the source tree.  You have to install to the `${CMAKE_INSTALL_PREFIX}` in order for the executables and libraries to become part of the package.  The easiest way I can figure out to do this is to see whether a `-DCMAKE_INSTALL_PREFIX` has been passed to `cmake` (or if it's still the default).  From the conda-forge recipe, I call:
```
cmake -DCMAKE_INSTALL_PREFIX=${PREFIX} -DCMAKE_BUILD_TYPE=BUILD_MODE ..
```
whereas your recommended recipe excludes the first flag, and thus should still build locally.

File: uni/CMakeLists.txt [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+else ()
+  find_library (LIBJIGSAW jigsaw HINTS "${CMAKE_INSTALL_PREFIX}/lib")
+endif ()
 
 message ("libjigsaw location: ${LIBJIGSAW}")
 
 add_executable (test_1 test_1.c)
 target_link_libraries (test_1 ${LIBJIGSAW})
-set_target_properties(test_1 PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE) 
+set_target_properties(test_1 PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)
```
Comment (xylar) [inline uni/CMakeLists.txt]: My editor automatically cleans up trailing white space...

**Comment by xylar** [comment]
With these changes, I'm able to build successfully for conda-forge and should be good to go once you make the `0.9.10` release.

**Comment by dengwirda** [comment]
Thanks @xylar -- I'm going to add a few more features to `0.9.10` before releasing, but it's hopefully fairly close now. Over the next few days for sure.

**Pull request merged**